### ISSUE TITLE
structcheck: Invalid Unused Field Warning

### ISSUE THREAD

**Issue by markvincentcaro** [golangci/golangci-lint#909]
Sample Code:

**_main.go_**

```
package main

import (
    "fmt"
)

type Dog struct {
    name string
}

type Dalmatian Dog

func main() {
    var dalmatian Dalmatian
    dalmatian.name = "spot"
    fmt.Println(dalmatian)
}
```

**_Expected Result:_**

Because Dog.name is used by Dalmatian, the field should not be flagged as unused.

**_Actual Result:_**

Dog.name is incorrectly flagged as unused even though it is being used by Dalmatian.

**_Warning Text:_**

```
`name` is unused (structcheck) go-lint
```

**_Environment:_** Windows 10 64-bit

**_IDE:_** Visual Studio Code

**_golangci-lint version:_** Latest as of January 5, 2020.

**Comment by fastest963** [comment]
I'm having a similar issue. `staticcheck` complains about all of the fields in `ourResponseWriterMetadata` even though I'm using all of them.
Here's an example of a few places where we use them:

```go
type ourResponseWriterMetadata struct {
	status     int
	skip       bool
	supports   int8
	written    int64
	encoding   string
	firstWrite time.Time
}

type ourResponseWriter struct {
	http.ResponseWriter
	ourResponseWriterMetadata
	enc writeCloserFlusher
	buf []byte
}

func (w *ourResponseWriter) Write(b []byte) (int, error) {
	if w.firstWrite.IsZero() {
		w.firstWrite = time.Now()
	}
        ...
}

func (w *ourResponseWriter) WriteHeader(code int) {
	w.status = code
}
```
